src/falconpy/sensor_download.py

```diff
@@ -0,0 +1,90 @@
+from ._util import service_request, parse_id_list
+from ._service_class import ServiceClass
+
+import os
+
+
+class Sensor_Download(ServiceClass):
+
+    def GetCombinedSensorInstallersByQuery(self: object, params: dict) -> dict:
+        """
+        retrieve all metadata for installers from provided query
+        """
+        FULL_URL = self.base_url+'/sensors/combined/installers/v1'
+        HEADERS = self.headers
+        PARAMS = params
+        returned = service_request(caller=self,
+                                   method="GET",
+                                   endpoint=FULL_URL,
+                                   params=PARAMS,
+                                   headers=HEADERS,
+                                   verify=self.ssl_verify
+                                   )
+        return returned
+
+    def DownloadSensorInstallerById(self: object, _id: str, file_name: str, download_path: str = "sensor_downloads"):
+        """
+        download the sensor by the sha256 into the specified directory.
+        the path will be created for the user if it does not already exist
+        """
+        # create the directory if doesn't exist
+        os.makedirs(download_path, exist_ok=True)
+        # _id is the sha256 of the sensor
+        FULL_URL = self.base_url+"/sensors/entities/download-installer/v1?id={}".format(_id)
+        HEADERS = self.headers
+        returned = service_request(caller=self,
+                                   method="GET",
+                                   endpoint=FULL_URL,
+                                   headers=HEADERS,
+                                   verify=self.ssl_verify
+                                   )
+        # write the newly downloaded sensor into the aforementioned directory with provided file name
+        with open(os.path.join(download_path, file_name), "wb") as sensor:
+            sensor.write(returned)
+        return True if returned else False
```

Comment:
```suggestion
        if download_path:
             # create the directory if doesn't exist
             os.makedirs(download_path, exist_ok=True)
             # write the newly downloaded sensor into the aforementioned directory with provided file name
             with open(os.path.join(download_path, file_name), "wb") as sensor:
                    sensor.write(returned)
                    # Let's add a "successfully downloaded" response here, below does not match pattern just an example
                    returned = { "message": "Download successful" }
        
        return returned
```

Reply:
I don't think we should have the library downloading files by default.  The default functionality should match other classes which is to return the object as it's received from the API.

That being said, this is really cool, and I bet a lot of folks would use it this way, so let's keep it as an optional variation.

We need to craft a response for the developer.  We might get away with using generate_error_result and sending a 200 for the code.

Reply:
Spacing might be borked on this suggestion, the GH editor wouldn't let me use tab...

Reply:
updated this a bit. may want to request API team to return a status code along with the downloads. for now checking that user passes a filename, download directory and the request returns the bytes.